### ISSUE TITLE
Subprocesses cleanup breaks too easily

### ISSUE THREAD

**Issue by ark3** [telepresenceio/telepresence#586]
This happens sometimes in `test_disconnect`.

```
Traceback (most recent call last):
  File "/home/circleci/project/telepresence/cli.py", line 74, in call_f
    return f(*args, **kwargs)
  File "/home/circleci/project/telepresence/main.py", line 507, in go_too
    runner, remote_info, args, env, subprocesses, socks_port, ssh
  File "/home/circleci/project/telepresence/local.py", line 166, in run_local_command
    wait_for_exit(runner, p, subprocesses)
  File "/home/circleci/project/telepresence/cleanup.py", line 84, in wait_for_exit
    dead_process = processes.any_dead()
  File "/home/circleci/project/telepresence/cleanup.py", line 60, in any_dead
    self.killall()
  File "/home/circleci/project/telepresence/cleanup.py", line 46, in killall
    killer()
  File "/home/circleci/project/telepresence/cleanup.py", line 38, in kill
    kill_process(process)
  File "/home/circleci/project/telepresence/cleanup.py", line 13, in kill_process
    process.terminate()
  File "/opt/circleci/.pyenv/versions/3.5.2/lib/python3.5/subprocess.py", line 1782, in terminate
    self.send_signal(signal.SIGTERM)
  File "/opt/circleci/.pyenv/versions/3.5.2/lib/python3.5/subprocess.py", line 1777, in send_signal
    os.kill(self.pid, sig)
ProcessLookupError: [Errno 3] No such process
```

**Comment by ark3** [comment]
More generally, any sort of crash during subprocesses cleanup breaks cleanup of the remaining subprocesses.

**Comment by ark3** [comment]
Fixed by #853; I botched the commit message.
